beacon-chain/p2p/options.go

```diff
@@ -31,16 +31,17 @@ func MultiAddressBuilder(ipAddr string, port uint) (ma.Multiaddr, error) {
 }
 
 // buildOptions for the libp2p host.
-func (s *Service) buildOptions(ip net.IP, priKey *ecdsa.PrivateKey) []libp2p.Option {
+func (s *Service) buildOptions(ip net.IP, priKey *ecdsa.PrivateKey) ([]libp2p.Option, error) {
 	cfg := s.cfg
 	listen, err := MultiAddressBuilder(ip.String(), cfg.TCPPort)
 	if err != nil {
 		log.WithError(err).Fatal("Failed to p2p listen")
 	}
 	if cfg.LocalIP != "" {
 		if net.ParseIP(cfg.LocalIP) == nil {
-			log.Fatalf("Invalid local ip provided: %s", cfg.LocalIP)
+			return nil, errors.Wrapf(err, "invalid local ip provided: %s", cfg.LocalIP)
```

Comment:
every other error is a fatal log here, it would be better to either return an error for all or have fatal logs for all.